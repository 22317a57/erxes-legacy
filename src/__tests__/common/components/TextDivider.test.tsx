import { mount, shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import TextDivider from '../../../modules/common/components/TextDivider';

describe('TextDivider component', () => {
  const defaultProps = {
    text: 'today',
    date: new Date()
  };

  test('renders TextDivider successfully', () => {
    shallow(<TextDivider {...defaultProps} />);
  });

  test('renders successfully with default value', () => {
    const wrapper = mount(<TextDivider {...defaultProps} />);
    const props = wrapper.props();

    expect(props).toMatchObject(defaultProps);
  });
  test('snapshot matches', () => {
    const rendered = renderer
      .create(<TextDivider {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
