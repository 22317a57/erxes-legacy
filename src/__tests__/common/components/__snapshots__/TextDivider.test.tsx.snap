// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextDivider component snapshot matches 1`] = `
<div
  className="sc-keFjpB kXdFqZ"
>
  <span>
    today
    <footer
      onBlur={[Function]}
      onClick={null}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      1:26 PM
    </footer>
  </span>
</div>
`;
